common/src/main/java/io/druid/common/config/Log4jShutdown.java

```diff
@@ -138,4 +140,55 @@ public boolean isStopped()
   {
     return State.STOPPED.equals(getState());
   }
+
+  private static class StateHolder
```

Comment:
if this ever moves out of this class it should probably be renamed `SynchronizedStateHolder`